CommonTools/Egamma/src/EffectiveAreas.cc

```diff
@@ -40,6 +44,48 @@ EffectiveAreas::EffectiveAreas(const std::string& filename) : filename_(filename
   checkConsistency();
 }
 
+EffectiveAreas::EffectiveAreas(const std::string& filename, const bool quadraticEAflag)
```

Comment:
Shouldn't this constructor also be able to handle the case of `quadraticEAflag=false`, e.g. by calling the other constructor?

Reply:
Yes, indeed. But, still I kept the previous code unchanged. I can delete the other constructor if you suggest so.

Reply:
It seems you still want to support the case where `quadraticEAflag` is `false`, but right now calling `EffectiveAreas(..., false)` does not produce the right behaviour.
What you could to is put the code from the first constructor into a `LoadLegacyEAFile` method and the code from the second one into a `LoadQuadraticEAFile` method. Then, the first constructor always calls `LoadLegacyEAFile`, while the second one calls the right method depending on the value of `quadraticEAflag`.

Reply:
I was unable to add my response to your latest comment on this, so replying here. I have removed the old constructor and made `quadraticEAflag=false` the default value in the new constructor. So, any portion of the existing code calling this constructor will now produce the correct behaviour.